### ISSUE TITLE
[BUG] Fatal error: Unexpectedly found nil while unwrapping an Optional value in getSortedLocalizations() (iOS 17)

### ISSUE THREAD

**Issue by tkafka** [superwall-me/Superwall-iOS#144]
## New issue checklist
- [X] I have reviewed the [`README`](https://github.com/superwall-me/paywall-ios/blob/master/README.md) and [documentation](https://docs.superwall.com/docs)
- [X] I have searched [existing issues](https://github.com/superwall-me/paywall-ios/issues) and this is not a duplicate
- [ ] I have attempted to reproduce the issue and include an example project.

### General information

- `Superwall` version: 3.0.1
- iOS version(s): 17.0 beta 1
- CocoaPods/Carthage version (if applicable):
- Xcode version: 15.0 beta 1
- Devices/Simulators affected: iOS 17 simulator
- Reproducible in the demo project? (Yes/No):
- Related issues:

### Describe the bug
I have built and ran an app with Superwallkit 3.0.1 in Xcode 15 for iOS 17 simulator. The app crashed with `Thread 1: Fatal error: Unexpectedly found nil while unwrapping an Optional value` on `LocalizationLogic.swift:22` while unpacking the `currentLocale.localizedString(forLanguageCode: localeId)` when `localeId` = `bgc`.

It seems to be that either there should not be a need for Superwall to enumerate hundreds of locales, or that the enumeration should be coded defensively, with an `if let` instead of force unwrap.

### Other Information
<img width="1178" alt="image" src="https://github.com/superwall-me/Superwall-iOS/assets/876717/91ec13eb-100e-43e0-958c-e566b64e2b80">

Scheme settings:
<img width="958" alt="image" src="https://github.com/superwall-me/Superwall-iOS/assets/876717/b0a9f0c2-83e1-4598-9a2c-a9485dcf1336">

**Comment by tkafka** [comment]
A repro code. When `Locale.autoupdatingCurrent` is `cs_001`, the `currentLocale.localizedString(forLanguageCode: localeId)` returns nil for `bgc` localeId.

When I do the same with `localeCs = NSLocale(localeIdentifier: "cs_001")`, it works.

So, Superwall enumeration now crashes the whole app on devices with Czech locale on iOS 17. I reported this to Apple, but this needs to be changed into a defensive code.

```swift
	func printLocales() {
		let localeIds = Locale.availableIdentifiers
		
		let currentLocale = Locale.autoupdatingCurrent
		let localeUs = NSLocale(localeIdentifier: "en_US")
		let localeCs = NSLocale(localeIdentifier: "cs_001")
		
		print("localeIdentifier, Current (\(currentLocale.identifier)), US, CS")
		localeIds.forEach { localeId in
			/// Get language
			let localizedLanguageCurrentLocale = currentLocale.localizedString(
				forLanguageCode: localeId
			)
			let localizedLanguageUs = localeUs.localizedString(
				forLanguageCode: localeId
			)
			let localizedLanguageCs = localeCs.localizedString(
				forLanguageCode: localeId
			)
			
			print("\(localeId),\"\(localizedLanguageCurrentLocale ?? "NO LOCALIZED STRING IN CURRENT LOCALE (\(currentLocale.identifier))")\",\"\(localizedLanguageUs ?? "NO US LOCALIZED STRING")\",\"\(localizedLanguageCs ?? "NO CS LOCALIZED STRING")\"")
		}
	}
```

**Comment by tkafka** [comment]
Guys, I looked at how much `swiftlint:disable:next force_unwrapping` is used in Superwallkit.
`!` definitely feels like a code smell to me - each place is a potential crash site for the app.

**Comment by yusuftor** [comment]
Thanks for this report, I've fixed this on the develop branch now. I've also moved the localization manager to be lazily loaded only when debugging paywalls, which means that we now won't loop through the localizations every time the app is launched from a cold start. Good point about the disabling of `force_unwrapping`, I've reduced these instances as well as `force_cast`.

**Comment by tkafka** [comment]
Thanks, this definitely happens for multiple languages, as I had crash reports from multiple people around the world. Will be happy to move back to official SDK once this is merged into some release.

**Comment by yusuftor** [comment]
Thanks, we've had some reports from others too since. Again keeping this open for visibility until it's merged into the next release.

**Comment by yusuftor** [comment]
Released in 3.0.2!
